src/goCheck.ts

```diff
@@ -29,11 +29,16 @@ export function removeTestStatus(e: vscode.TextDocumentChangeEvent) {
 	statusBarItem.text = '';
 }
 
-export function notifyIfGeneratedFile(this: void, e: vscode.TextDocumentChangeEvent) {
+export function checksOnFileEdit(this: void, e: vscode.TextDocumentChangeEvent) {
 	const ctx: any = this;
 	if (e.document.isUntitled || e.document.languageId !== 'go') {
 		return;
 	}
+	notifyIfGeneratedFile(ctx, e);
+	notifyOnBuildTagMismatch(ctx, e);
```

Comment:
What is the reason to have this notification be shown as part of the file change event?
I would expect the check on build tags to happen right before the features mentioned in #1497 are triggered.
Otherwise one would get prompted way more times and even in situations where they might be doing simple edits like renaming or formatting and not really looking to build the code.